### DISCUSSION TITLE
Why blocking in same routers?

### DISCUSSION THREAD

**Discussion by VladimirShestakov** [tokio-rs/axum#2321] [Q&A] [answered]
Why does a request to route /api/1 block second request to the /api/1 route? At the same time, does not block request to route /api/2. That is, requests for the same routes queue up. How to fix bloking in same routes?

```rs
#[tokio::main]
async fn main() {
  let app = Router::new()
    .route("/api/1", get(handler))
    .route("/api/2", get(handler));

  axum::Server::bind(&"127.0.0.1:5000".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
}

async fn handler() -> impl IntoResponse {
  const MESSAGE: &str = "Build Simple CRUD API in Rust using Axum";

  let json_response = serde_json::json!({
    "status": "success",
    "message": MESSAGE
  });
 
  // emulate a long wait
  tokio::time::sleep(Duration::from_millis(5000)).await;

  Json(json_response)
}
```

axum = "0.6.20"
tokio = { version = "1.34.0", features = ["full"] }
windows10

**Comment by jplatte** [comment, marked as answer]
How are you testing this? It's likely a client-side synchronization, not the server being unable to handle both requests in parallel.

**Reply by VladimirShestakov** [reply]
I'm sorry, everything is really parallel
